pkg/kubelet/container/runtime_cache.go

```diff
@@ -80,6 +80,14 @@ func (r *runtimeCache) GetPods() ([]*Pod, error) {
 	return r.pods, nil
 }
 
+// getPods returns the current cached pods. This is used mainly for testing
+// because it does not start the refreshing thread.
+func (r *runtimeCache) getCachedPods() []*Pod {
+	r.Lock()
+	defer r.Unlock()
```

Comment:
I just realized now that I forgot to include the test that I wrote :(

To answer your question, this function is purely used to verify the content of the cache after performing certain operations. If you call GetPods once at anytime in your test, the update thread could still be active, and you cannot safely retrieve the list without locking anymore.

In fact, in my test it'd work without this lock, but it's a preventive measure so that when more tests are added, we wouldn't get random data race showing up on travis/shippable.